internal/service/searchdeployment/state_transition_search_deployment_test.go

```diff
@@ -3,155 +3,126 @@ package searchdeployment_test
+	sc500    = conversion.IntPtr(500)
+	sc503    = conversion.IntPtr(503)
+)
+
+type testCase struct {
+	expectedState *string
+	name          string
+	mockResponses []response
+	expectedError bool
 }
 
 func TestSearchDeploymentStateTransition(t *testing.T) {
-	testCases := []stateTransitionTestCase{
+	testCases := []testCase{
 		{
 			name: "Successful transition to IDLE",
-			mockResponses: []SearchDeploymentResponse{
-				{
-					DeploymentResp: responseWithState("UPDATING"),
-				},
-				{
-					DeploymentResp: responseWithState("IDLE"),
-				},
+			mockResponses: []response{
+				{state: &updating},
+				{state: &idle},
 			},
-			expectedResult: responseWithState("IDLE"),
-			expectedError:  false,
+			expectedState: &idle,
+			expectedError: false,
 		},
 		{
 			name: "Successful transition to IDLE with 503 error in between",
-			mockResponses: []SearchDeploymentResponse{
-				{
-					DeploymentResp: responseWithState("UPDATING"),
-				},
-				{
-					DeploymentResp: nil,
-					HTTPResponse:   &http.Response{StatusCode: 503},
-					Err:            errors.New("Service Unavailable"),
-				},
-				{
-					DeploymentResp: responseWithState("IDLE"),
-				},
+			mockResponses: []response{
+				{state: &updating},
+				{statusCode: sc503, err: errors.New("Service Unavailable")},
+				{state: &idle},
 			},
-			expectedResult: responseWithState("IDLE"),
-			expectedError:  false,
+			expectedState: &idle,
+			expectedError: false,
 		},
 		{
 			name: "Error when transitioning to an unknown state",
-			mockResponses: []SearchDeploymentResponse{
-				{
-					DeploymentResp: responseWithState("UPDATING"),
-				},
-				{
-					DeploymentResp: responseWithState(""),
-				},
+			mockResponses: []response{
+				{state: &updating},
+				{state: &unknown},
 			},
-			expectedResult: nil,
-			expectedError:  true,
+			expectedState: nil,
+			expectedError: true,
 		},
 		{
 			name: "Error when API responds with error",
-			mockResponses: []SearchDeploymentResponse{
-				{
-					DeploymentResp: nil,
-					HTTPResponse:   &http.Response{StatusCode: 500},
-					Err:            errors.New("Internal server error"),
-				},
+			mockResponses: []response{
+				{statusCode: sc500, err: errors.New("Internal server error")},
 			},
-			expectedResult: nil,
-			expectedError:  true,
+			expectedState: nil,
+			expectedError: true,
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			mockService := MockSearchDeploymentService{
-				MockResponses: tc.mockResponses,
-			}
-
-			resp, err := searchdeployment.WaitSearchNodeStateTransition(context.Background(), dummyProjectID, "Cluster0", &mockService, testTimeoutConfig)
-
-			if (err != nil) != tc.expectedError {
-				t.Errorf("Case %s: Received unexpected error: %v", tc.name, err)
-			}
-
-			if !reflect.DeepEqual(tc.expectedResult, resp) {
-				t.Errorf("Case %s: Response did not match expected output", tc.name)
+			svc := mocksvc.NewDeploymentService(t)
```

Comment:
nit: 
```suggestion
			mockSvc := mocksvc.NewDeploymentService(t)
```

just so it's clear below that svc is mocked

Reply:
I preferred to keep svc because I think it's clear in this context, but I can change it

Reply:
ok fair, in the end you can see below it is being used with `.On` comments. I don't have big concerns here